### ISSUE TITLE
After encoded to tfrecords and decoded, .jpg quality descend

### ISSUE THREAD

**Issue by czx94** [tensorflow/tensorflow#19830]
I'm not sure if somebody has noticed or it's just an exception that after the procedure of encoding and decoding, the .jpg loss some information. I havn't figured it out.... So strange..........
Hope someone can find the answer...

The part to encode:
```
def encode_tfrecords(file_path, output_path):
    classes = os.listdir(file_path)

    with tf.python_io.TFRecordWriter(output_path) as writer:
        for index, name in enumerate(classes):
            class_path = file_path + name + "/"
            for img_name in os.listdir(class_path):
                img_path = class_path + img_name

                img_raw = tf.gfile.FastGFile(img_path, 'rb').read()

                example = tf.train.Example(features=tf.train.Features(feature={
                    "label": tf.train.Feature(int64_list=tf.train.Int64List(value=[index])),
                    'img_raw': tf.train.Feature(bytes_list=tf.train.BytesList(value=[img_raw]))
                }))
                writer.write(example.SerializeToString())  
```

The part to decode:
```
def read_and_decode(filename, output_path, iters):
    filename_queue = tf.train.string_input_producer([filename])  
    reader = tf.TFRecordReader()
    _, serialized_example = reader.read(filename_queue) 
    features = tf.parse_single_example(serialized_example,
                                       features={
                                           'label': tf.FixedLenFeature([], tf.int64),
                                           'img_raw': tf.FixedLenFeature([], tf.string),
                                       }) 
    # image = tf.decode_raw(features['img_raw'], t.uint8)
    image = tf.image.decode_jpeg(features['img_raw'], channels=3)
    label  = tf.cast(features['label'], tf.int64)
    with tf.Session() as sess:  
        init_op = tf.global_variables_initializer()
        sess.run(init_op)
        coord = tf.train.Coordinator()
        threads = tf.train.start_queue_runners(coord=coord)
        for i in range(iters):
            example, l = sess.run([image, label]) 
            img = Image.fromarray(example, 'RGB')
            img.save(output_path + str(i) + '_''Label_' + str(l) + '.jpg')
        coord.request_stop()
        coord.join(threads)
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by czx94** [comment]
Thx man!
It's executed in my mac OS with py 3.6.5 and tf 1.3.0. 
To reproduce, you should have a directory we call file_path of such structure below:
0 -- img1.jpg
     img2.jpg
     img3.jpg
     ...
1 -- img1.jpg
     img2.jpg
     ...
2 -- ...
And encode_tfrecords'll return a tfrecords file.
Then you can use read_and_decode function to decode the tfrecords and see the pics.

**Comment by czx94** [comment]
And I have another question as well, I noticed that in tf for encoding we have tf.train.BytesList, tf.train.Int64List, tf.train.FloatList, but what if we want to encode something string like. Is there any solution? I found that encoding to Bytes didn't work well

**Comment by drpngx** [comment]
What is the problem? jpg is lossy? You can use png instead.

**Comment by drpngx** [comment]
png should be lossless. jpg is lossy, by design.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
